weather_mv/loader_pipeline/sinks.py

```diff
@@ -72,15 +73,41 @@ def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
     return data_ds
 
 
-def _preprocess_tif(ds: xr.Dataset, filename: str, tif_metadata_for_datetime: str) -> xr.Dataset:
+def match_datetime(file_name: str, regex_str: str) -> datetime.datetime:
+    """Extracts the datetime object from a string
+    """
+    char_to_replace = {
+        '%Y': '([0-9]{4})',
+        '%m': '([0-9]{2})',
+        '%d': '([0-9]{2})',
+        '%H': '([0-9]{2})',
+        '%M': '([0-9]{2})',
+        '%S': '([0-9]{2})',
+        '*': '.*'
```

Comment:
Do we have to use a regex parser here? Can we use the default datetime parser in python instead?